programs/bpf_loader/src/syscalls.rs

```diff
@@ -1679,12 +1679,16 @@ fn check_account_infos(
 fn check_authorized_program(
     program_id: &Pubkey,
     instruction_data: &[u8],
+    invoke_context: &Ref<&mut dyn InvokeContext>,
 ) -> Result<(), EbpfError<BpfError>> {
     if native_loader::check_id(program_id)
         || bpf_loader::check_id(program_id)
         || bpf_loader_deprecated::check_id(program_id)
         || (bpf_loader_upgradeable::check_id(program_id)
-            && !bpf_loader_upgradeable::is_upgrade_instruction(instruction_data))
+            && !(bpf_loader_upgradeable::is_upgrade_instruction(instruction_data)
```

Comment:
Is this cascading condition ok should I extract it into a helper function?

Reply:
I'm fine with this, still very readable in my opinion

Reply:
After the feature is active for > 2 epochs on master we can remove the featurization